racingcar/racingcar.go

```diff
@@ -1,6 +1,69 @@
 package racingcar
 
-// 예시 함수, 구현 시 제거
-func Hello() string {
-	return "Hello, Racing Car!"
+import (
+	"fmt"
+	"math/rand"
+	"strings"
+)
+
+type CarRace struct {
+	CarNames []string
+	CarsInfo map[string]int
+	Count    int
+}
+
+func NewCarRace(carNames []string, count int) *CarRace {
+	carsInfo := make(map[string]int)
+	for _, name := range carNames {
+		carsInfo[strings.TrimSpace(name)] = 0
+	}
+
+	return &CarRace{
+		CarNames: carNames,
+		CarsInfo: carsInfo,
+		Count:    count,
+	}
+}
+
+func (c *CarRace) Start() []string {
+	fmt.Println("실행 결과")
+	for i := 0; i < c.Count; i++ {
+		for carName := range c.CarsInfo {
+			if rand.Float64() < 0.5 {
+				c.CarsInfo[carName]++
+			}
+			fmt.Println(carName, " : ", strings.Repeat("-", c.CarsInfo[carName]))
+		}
```

Comment:
아래서 하신것처럼 carNmae이랑 position을 같이 받아서 사용해도 될 것 같아요

```suggestion
		for carName, position := range c.CarsInfo {
			if rand.Float64() < 0.5 {
				c.CarsInfo[carName]++
			}
			fmt.Println(carName, " : ", strings.Repeat("-", position))
		}

```